shared/cm_string.h

```diff
@@ -213,19 +213,3 @@ int stricmp(view str1, view str2);
 //------------------------------------------------------------------------------
 string::view va(string::literal fmt, ...);
 
-////////////////////////////////////////////////////////////////////////////////
-namespace std {
-
-template<class> struct less;
-
-//------------------------------------------------------------------------------
-template<> struct less<::string::buffer>
```

Comment:
This allows `std::map<string::buffer, ...>` to be searched using `string::view`, so disabling this code is why `cSound::load_sound` below doesn't compile. Not sure why Apple Clang doesn't like this code, maybe the declaration of `template<class> struct less` doesn't match the one in their standard library?

Reply:
Errors are:

```
/Users/liz/Documents/quark/shared/./cm_string.h:222:23: error: explicit specialization of non-template struct 'less'
    template<> struct less<::string::buffer>

/Users/liz/Documents/quark/shared/./cm_string.h:222:23: error: redefinition of 'less' as different kind of symbol
/Users/liz/Documents/quark/shared/./cm_string.h:219:28: note: previous definition is here
    template<class> struct less;
```

I'll take time to look into it! I'm not very good at c/cpp so I avoided the problem (read: templates) just to get to a successful build, but now that we're here why not :^)

Reply:
I think I understand templates and template specialization.

The existing impl seems right though, so I dunno? :s There's this but it seems about the same I think: https://stackoverflow.com/a/43648400

There doesn't seem to be anything special about `less` in terms of definition as far as I can tell.

Reply:
I'm at a loss on this one. The error says it's not a template, but it's declared as a template right there. Can't find anything on the internet to explain it and can't reproduce without AppleClang. :/

Reply:
We'll figure it out when you're back ᕕ( ᐛ )ᕗ

Reply:
Yeah, can leave as-is for now.